### PR TITLE
aurora-client cli

### PR DESCRIPTION
**Pull request by thinker0**
# Apache Aurora Scheduler

cli tools
https://aurora.apache.org/

### CODE REVIEW

File: Library/Formula/aurora-scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class AuroraScheduler < Formula
+  desc "Apache Aurora Scheduler Client"
+  homepage "http://aurora.apache.org/"
+  url "http://www.us.apache.org/dist/aurora/0.9.0/apache-aurora-0.9.0.tar.gz"
+  version "0.9.0"
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: Won't need this line.

File: Library/Formula/aurora-scheduler.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on "python" => :build
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: Should be either `depends_on :python => :build` or `depends_on :python => :build if MacOS.version <= :snow_leopard`, I believe. @tdsmith

File: Library/Formula/aurora-scheduler.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class AuroraScheduler < Formula
+  desc "Apache Aurora Scheduler Client"
+  homepage "http://aurora.apache.org/"
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: Should be able to use `https://` here.

File: Library/Formula/aurora-scheduler.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  url "http://www.us.apache.org/dist/aurora/0.9.0/apache-aurora-0.9.0.tar.gz"
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: This needs to use the `https://www.apache.org/dyn/closer.cgi?path=` style URL.

File: Library/Formula/aurora-scheduler.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  version "0.9.0"
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on "python" => :build
+
+  def install
+    system "./pants", "binary", "src/main/python/apache/aurora/admin:kaurora_admin"
+    system "./pants", "binary", "src/main/python/apache/aurora/client/cli:kaurora"
+    system "mv", "dist/kaurora.pex", "dist/aurora"
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: You can combine these four lines into:

```ruby
bin.install "dist/kaurora.pex" => "aurora"
bin.install "dist/kaurora_admin.pex" => "aurora_admin"
```

File: Library/Formula/aurora-scheduler.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    system "mv", "dist/kaurora_admin.pex", "dist/aurora_admin"
+    bin.install "dist/aurora"
+    bin.install "dist/aurora_admin"
+  end
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. e.g. write out a `relx.config` and do something with it. Thanks!

**Comment by thinker0** [comment]
@DomT4 ? merge schedule ?

File: Library/Formula/aurora-scheduler.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class AuroraScheduler < Formula
+  desc "Apache Aurora Scheduler Client"
+  homepage "https://aurora.apache.org/"
+  url "https://www.apache.org/dyn/closer.cgi?path=aurora/0.9.0/apache-aurora-0.9.0.tar.gz"
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on :python => :build
```
Comment (tdsmith) [inline Library/Formula/aurora-scheduler.rb]: You can remove this line unless Python 2.6 won't work.

File: Library/Formula/aurora-scheduler.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    ENV["CFLAGS"] = "-Qunused-arguments"
+    ENV["CPPFLAGS"] = "-Qunused-arguments"
+    system "./pants", "binary", "src/main/python/apache/aurora/client/cli:kaurora"
+    system "./pants", "binary", "src/main/python/apache/aurora/admin:kaurora_admin"
+    bin.install "dist/kaurora.pex" => "aurora"
+    bin.install "dist/kaurora_admin.pex" => "aurora_admin"
+  end
+
+  test do
+    system "aurora", "--version"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Can’t we write a test that does something more substantial than e.g. `--version`/`--help`?

File: Library/Formula/aurora-scheduler.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+class AuroraScheduler < Formula
+  desc "Apache Aurora Scheduler Client"
+  homepage "https://aurora.apache.org/"
+  url "https://www.apache.org/dyn/closer.cgi?path=aurora/0.9.0/apache-aurora-0.9.0.tar.gz"
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on :python => :build
+
+  def install
+    ENV["LC_ALL"] = "en_US.UTF-8"
```
Comment (DomT4) [inline Library/Formula/aurora-scheduler.rb]: Why is this necessary, out of curiosity?

Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: some time, my environment
```
LC_ALL=UTF.8
```
virtual env Python Pants not build, not work.

**Comment by thinker0** [comment]
@bfontaine @DomT4 @tdsmith 

aurora, aurora_admin 

CLI ':test' vary hard problem

- 1 zookeeper Server
- 2 mesos-master instance
- 3 mesos-slave instance
- 4 aurora-scheduler (Main Daemon)
- 5 aurora-executer (Slafe Daemon)

http://bot.brew.sh/job/Homebrew%20Pull%20Requests/33078/ 

```
Failed mavericks Failed mountain_lion Failed yosemite 
```

**Comment by thinker0** [comment]
i wan`t **aurora**, **aurora_admin** cli tool on brew

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

File: Library/Formula/aurora-scheduler.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    bin.install "dist/kaurora_admin.pex" => "aurora_admin"
+    libexec.install "dist/thermos_executor.pex" => "bin/thermos_executor"
+    bin.install "dist/thermos_runner.pex" => "thermos_runner"
+    bin.install "dist/thermos_observer.pex" => "thermos_observer"
+  end
+
+  test do
+    mkdir_p("~/.aurora/")
+    touch("~/.aurora/clusters.json")
+    assert File.exist?("~/.aurora/clusters.json")
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: These lines will create a `~` directory in the testpath, which is probably not what you want.

Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: Thans.
currently aurora cluster configuration only location
* /etc/aurora/clusters.json
* ~/.aurora/clusters.json

you have solution?

**Comment by xu-cheng** [comment]
@thinker0 Can you test this locally before you submit a new commit? Too many commits kinda exhaust our CI resources.

**Comment by thinker0** [comment]
@xu-cheng yes
local very well nice working T.T.

**Comment by thinker0** [comment]
@xu-cheng 
can i help you 'gradle' error ?
```
  def install
    remove_file("buildSrc/build.gradle", true)
    touch("buildSrc/build.gradle")
    system "gradle", "wrapper", "--gradle-version", "2.6"
    system "./gradlew", "installDist"
    libexec.install Dir["dist/install/aurora-scheduler/*"]
 ...
```

**Comment by thinker0** [comment]
@bfontaine @DomT4 @tdsmith i`am finished ^^;

File: Library/Formula/aurora-scheduler.rb [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  def pour_bottle?
+    quiet_system("/usr/libexec/java_home --version 1.8 --failfast")
+  end
+
+  def install
+    ENV["PIP_DEFAULT_TIMEOUT"] = "60"
+    ENV["LC_ALL"] = "en_US.UTF-8"
+    ENV["CFLAGS"] = "-Qunused-arguments"
+    ENV["CPPFLAGS"] = "-Qunused-arguments"
+    File.write(".pantsversion", File.open(".pantsversion", &:read).gsub(/0\.0\.32/, "0.0.53"))
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Why do we need to do this?

File: Library/Formula/aurora-scheduler.rb [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+      jar_library(name = 'scala-repl',
+                  jars = [
+                    jar(org = 'org.scala-lang', name = 'jline', rev = SCALA_REV, intransitive = True),
+                  ],
+                  dependencies = [
+                    ':scala-compiler',
+                    ':scala-library',
+                  ])
+    EOS
+    File.open("BUILD.tools", "a+") { |f| f.puts(tools) }
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Why is this necessary?

File: Library/Formula/aurora-scheduler.rb (hunk 11 above)
Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: Issue : https://issues.apache.org/jira/browse/AURORA-1508

pants 0.0.53 : El Capitan Build Environment.

File: Library/Formula/aurora-scheduler.rb (hunk 12 above)
Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: El Capitan : pants 0.0.53
work in https://github.com/pantsbuild/pants/blob/master/BUILD.tools

File: Library/Formula/aurora-scheduler.rb (hunk 11 above)
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Use `inreplace ".pantsversion", "0.0.32", "0.0.53"` and add this URL as a comment above it.

Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: :+1:

File: Library/Formula/aurora-scheduler.rb (hunk 12 above)
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: It’s probably better to write this as a `:DATA` patch with this link in a comment above it.

Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]:
```
:DATA patch
```
?... any example please ^^

File: Library/Formula/aurora-scheduler.rb [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+  def pour_bottle?
+    quiet_system("/usr/libexec/java_home --version 1.8 --failfast")
+  end
+
+  def install
+    ENV["PIP_DEFAULT_TIMEOUT"] = "60"
+    ENV["LC_ALL"] = "en_US.UTF-8"
+    ENV["CFLAGS"] = "-Qunused-arguments"
+    ENV["CPPFLAGS"] = "-Qunused-arguments"
+    inreplace ".pantsversion", "0.0.32", "0.0.53"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Please add a comment above this line with the upstream issue’s URL (https://issues.apache.org/jira/browse/AURORA-1508) so we know why it’s there.

File: Library/Formula/aurora-scheduler.rb [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on :java => "1.8+"
+
+  def pour_bottle?
+    quiet_system("/usr/libexec/java_home --version 1.8 --failfast")
+  end
+
+  def install
+    ENV["PIP_DEFAULT_TIMEOUT"] = "60"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Why is this? The build process doesn’t seem to use Python/Pip.

File: Library/Formula/aurora-scheduler.rb [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+    ENV["LC_ALL"] = "en_US.UTF-8"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Do we need this in order to build the formula?

File: Library/Formula/aurora-scheduler.rb [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+    ENV["CFLAGS"] = "-Qunused-arguments"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: `ENV.append_to_cflags "-Qunused-arguments"` should work here.

File: Library/Formula/aurora-scheduler.rb [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+    ENV["CPPFLAGS"] = "-Qunused-arguments"
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: `ENV.append "CPPFLAGS", "-Qunused-arguments"` should work here.

**Comment by thinker0** [comment]
@bfontaine unused remove

File: Library/Formula/aurora-scheduler.rb [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+  desc "Apache Aurora Scheduler Client"
+  homepage "https://aurora.apache.org"
+  url "https://www.apache.org/dyn/closer.cgi?path=aurora/0.9.0/apache-aurora-0.9.0.tar.gz"
+  sha256 "16040866f3a799226452b1541892eb80ed3c61f47c33f1ccb0687fb5cf82767c"
+
+  depends_on :java => "1.8+"
+
+  def pour_bottle?
+    quiet_system("/usr/libexec/java_home --version 1.8 --failfast")
+  end
```
Comment (bfontaine) [inline Library/Formula/aurora-scheduler.rb]: Why is this necessary?

Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: ok removed .

File: Library/Formula/aurora-scheduler.rb (hunk 16 above)
Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: ok removed .

File: Library/Formula/aurora-scheduler.rb (hunk 15 above)
Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: ok removed .

File: Library/Formula/aurora-scheduler.rb (hunk 14 above)
Comment (thinker0) [inline Library/Formula/aurora-scheduler.rb]: ok removed .

**Comment by thinker0** [comment]
@bfontaine any problem ?

**Comment by bfontaine** [comment]
@thinker0 According to upstream’s docs this depends on Java 1.8 and Python 2.7; so please add the following to the formula:
```rb
depends_on :java => "1.8+"
depends_on :python if MacOS.version <= :snow_leopard
```

**Comment by thinker0** [comment]
@bfontaine done
